source/parser/independent/symbol.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+* C++ Assembler Interpreter
+* Copyright (C) 2016 Chair of Computer Architecture
+* at Technical University of Munich
+*
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "parser/independent/symbol.hpp"
+#include "common/assert.hpp"
+
+// Some testing regexes.
+static const std::regex TRIMMED =
+    std::regex("(^\\s+|\\s+$)", std::regex_constants::optimize);
```

Comment:
Maybe use `R"(^\\s+|\\s+$)"`? This will not surround the regex with parentheses, but you don't need those here anyway